test/DataFixtures/ApiCalls/SelfServiceApisData.php

```diff
@@ -50,7 +51,7 @@ public function postApiCalls(): array
         $commentData = new CommentData;
 
         return [
-            'postComment' => [$commentData->getFilledInstance(), $commentData->getResponse()],
+            'postComment' => [(new CreateCommentData)->getFilledInstance(), $commentData->getResponse()],
```

Comment:
What's going on here? 🤔 what does `$commentData->getResponse()` do?

Reply:
the input to the test function data is a `CreateComment` object, the output is a `Comment` Object.
`getResponse` is essentially just the array data of a comment, wrapped in the API format, and that's what the API expects as an output inside the Request contents

in the data fixtures, we have:
1- `getArrayData`  which is basically the array format of any model and its child models.
2- `getResponse`, same above data, but wrapped in ApiFormat
3- `getAllResponse` same as the above, but the data is in format of array of objects
4- `getDataWithObjects` same as array data, but child models are objects not arrays
5- `getFilledInstaces` is essentially, $model->fill(getDataWithObjects());